internal/auth0/tenant/flatten.go

```diff
@@ -26,6 +25,13 @@ func flattenTenant(data *schema.ResourceData, tenant *management.Tenant) error {
 		data.Set("allow_organization_name_in_authentication_api", tenant.GetAllowOrgNameInAuthAPI()),
 	)
 
+	if tenant.GetIdleSessionLifetime() == 0 {
+		idleSessionLifetimeDefault := NewResource().Schema["idle_session_lifetime"].Default
+		result = multierror.Append(result, data.Set("idle_session_lifetime", idleSessionLifetimeDefault))
+	} else {
+		result = multierror.Append(result, data.Set("idle_session_lifetime", tenant.GetIdleSessionLifetime()))
+	}
+
```

Comment:
I would suggest we rework this a little bit as we're opening ourselves up for trouble by directly referencing the schema like this and then the properties' default value. Furthermore the `session_lifetime` is prone to the same issue as it has a default value set.

What about having these defaults as constants references inside the schema and then here. 

Example:
```
// within resource.go

const (
	idleSessionLifetimeDefault = 72
	sessionLifetimeDefault = 168
)

// within this flatten func
func flattenTenant(data *schema.ResourceData, tenant *management.Tenant) error {
	result := multierror.Append(
		data.Set("default_audience", tenant.GetDefaultAudience()),
		data.Set("default_directory", tenant.GetDefaultDirectory()),
		data.Set("default_redirection_uri", tenant.GetDefaultRedirectionURI()),
		data.Set("friendly_name", tenant.GetFriendlyName()),
		data.Set("picture_url", tenant.GetPictureURL()),
		data.Set("support_email", tenant.GetSupportEmail()),
		data.Set("support_url", tenant.GetSupportURL()),
		data.Set("allowed_logout_urls", tenant.GetAllowedLogoutURLs()),
		data.Set("session_lifetime", tenant.GetSessionLifetime()),
		data.Set("idle_session_lifetime", tenant.GetIdleSessionLifetime()),
		data.Set("sandbox_version", tenant.GetSandboxVersion()),
		data.Set("enabled_locales", tenant.GetEnabledLocales()),
		data.Set("flags", flattenTenantFlags(tenant.GetFlags())),
		data.Set("session_cookie", flattenTenantSessionCookie(tenant.GetSessionCookie())),
		data.Set("sessions", flattenTenantSessions(tenant.GetSessions())),
		data.Set("allow_organization_name_in_authentication_api", tenant.GetAllowOrgNameInAuthAPI()),
	)

	if tenant.GetIdleSessionLifetime() == 0 {
		result = multierror.Append(result, data.Set("idle_session_lifetime", idleSessionLifetimeDefault)
	}

	if tenant.GetSessionLifetime() == 0 {
		result = multierror.Append(result, data.Set("session_lifetime", idleSessionLifetimeDefault)
	}

	return result.ErrorOrNil()
}
```

Reply:
> we're opening ourselves up for trouble by directly referencing the schema like this and then the properties' default value

What trouble specifically? The default value is defined on the schema, it is _the_ source of truth. We can certainly abstract this value out as a const as a shortcut, but the default value itself should be defined as closely to the schema as possible IMO.

Reply:
We're instantiating the entire resource twice and then referencing a map and it's default value, that can be nil.